### DISCUSSION TITLE
Scalar compound attributes

### DISCUSSION THREAD

**Discussion by johnstairs** [BlueBrain/HighFive#598] [Questions]
Hi, it seems like scalar compound attributes are not supported. I'm wondering if that is intentional. 

For example, in the compound types [example](https://github.com/BlueBrain/HighFive/blob/a794a4d651b309069fec5c628d0fb82c2be92662/src/examples/compound_types.cpp#L42), if I call `createAttribute` with a single `Size2D` instead of `std::vector<Size2D>`, I get a compilation error:

> static assertion failed: supported datatype should be an arithmetic value, a std::string or a container/array

Of course I might be doing something wrong. (I also tried using `HighFive::DataSpace(HighFive::DataSpace::DataspaceType::dataspace_scalar)` but ended up with a runtime segfault)

Thanks for your help!

**Comment by ohm314** [comment]
Hi,

I took just a quick look. Here's what I tried: https://godbolt.org/z/1b4W5xfGf

and this seems to build fine. Does that help, or is it maybe a specific case that still doesn't work?

cheers!
